pkg/controller/config/config_controller.go

```diff
@@ -0,0 +1,305 @@
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource Network CRD
+	err = c.Watch(&source.Kind{Type: &configv1.Network{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileConfig implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileConfig{}
+
+// ReconcileConfig reconciles a ConfigMap object
+type ReconcileConfig struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+	status *statusmanager.StatusManager
+
+	appliedClusterConfig *configv1.Network
+	appliedOperConfig    *operatorv1.AntreaInstall
+}
+
+// Reconcile propagates changes from the cluster config and operater config to
+// antrea config. And then update antrea resources if antrea config changes.
+func (r *ReconcileConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+
+	if request.Namespace == "" && request.Name == operatortypes.ClusterConfigName {
+		reqLogger.Info("Reconciling antrea-operator Cluster Network CR change")
+	} else if request.Namespace == operatortypes.OperatorNameSpace && request.Name == operatortypes.OperatorConfigName {
+		reqLogger.Info("Reconciling antrea-operator antrea-install CR change")
+	} else {
+		return reconcile.Result{}, nil
+	}
+
+	// Fetch Cluster Network CR.
+	clusterConfig := &configv1.Network{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: operatortypes.ClusterConfigName}, clusterConfig)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			msg := "Cluster Network CR not found"
+			log.Info(msg)
+			r.status.SetDegraded(statusmanager.ClusterConfig, "NoClusterConfig", msg)
+			return reconcile.Result{}, nil
+		}
+		r.status.SetDegraded(statusmanager.ClusterConfig, "InvalidClusterConfig",
+			fmt.Sprintf("Failed to get cluster network CRD: %v", err))
+		log.Error(err, "failed to get Cluster Network CR")
+		return reconcile.Result{Requeue: true}, err
+	}
+
+	// Fetch antrea-install CR.
+	operConfig := &operatorv1.AntreaInstall{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: operatortypes.OperatorNameSpace, Name: operatortypes.OperatorConfigName}, operConfig)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			msg := fmt.Sprintf("%s CR not found", operatortypes.OperatorConfigName)
+			log.Info(msg)
+			r.status.SetDegraded(statusmanager.ClusterConfig, "NoAntreaInstallCR", msg)
+			return reconcile.Result{}, nil
+		}
+		log.Error(err, "failed to get antrea-install CR")
+		r.status.SetDegraded(statusmanager.OperatorConfig, "InvalidAntreaInstallCR",
+			fmt.Sprintf("Failed to get operator CR: %v", err))
+		return reconcile.Result{}, err
```

Comment:
The description of the function shows that the request will be requeued if returned error is non-nil. But I can still set `Requeue` as true to make it easy to read.
```
type Reconciler interface {
	// Reconciler performs a full reconciliation for the object referred to by the Request.
	// The Controller will requeue the Request to be processed again if an error is non-nil or
	// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
	Reconcile(Request) (Result, error)
}
```